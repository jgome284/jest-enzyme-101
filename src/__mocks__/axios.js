module.exports = {
    get: jest.fn(() => {
        return Promise.resolve({
            data: [
                {
                    id: 1,
                    name: 'Jane Doe',
                    email: 'janedoe@gmail.com',
                    username: 'jdoe'
                }
            ]
        })
    })
}