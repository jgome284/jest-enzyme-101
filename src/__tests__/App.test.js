import React from 'react'
import Enzyme, { shallow, render } from 'enzyme'
import axios from 'axios';
import App from '../App'

jest.mock('axios')

describe('<App />', () => {
    it("renders correctly using shallow", () => {
        const wrapper = shallow(
            <App />
        );
    
        expect(wrapper).toMatchSnapshot();
    });
    
    it("renders correctly using render", () => {
        const wrapper = render(
            <App />
        );
    
        expect(wrapper).toMatchSnapshot();
    });

    it('calls componentDidMount', () => {
        jest.spyOn(App.prototype, 'componentDidMount')
        const wrapper = shallow(<App />)
        expect(App.prototype.componentDidMount.mock.calls.length).toBe(1)
    })

    it('fetches a list of users', () => {
        const getSpy = jest.spyOn(axios, 'get')
        const wrapper = shallow(
            <App />
        )
        expect(getSpy).toBeCalled()
    })
})