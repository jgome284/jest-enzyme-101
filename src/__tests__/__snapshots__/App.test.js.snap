// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly using render 1`] = `
LoadedCheerio {
  "0": <div
    class="container"
  >
    <h2>
      React Testing with Jest and Enzyme
    </h2>
    <div
      class="row"
    >
      <p>
        Loading...
      </p>
    </div>
  </div>,
  "1": <div>
    <h2>
      Welcome on board!
    </h2>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

<<<<<<< HEAD
exports[`<App /> renders correctly using shallow 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="container"
      >
        <h2>
          React Testing with Jest and Enzyme
        </h2>
        <div
          className="row"
        >
          <p>
            Loading...
          </p>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            React Testing with Jest and Enzyme
          </h2>,
          <div
            className="row"
          >
            <p>
              Loading...
            </p>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "React Testing with Jest and Enzyme",
          },
          "ref": null,
          "rendered": "React Testing with Jest and Enzyme",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Loading...
            </p>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading...",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="container"
        >
          <h2>
            React Testing with Jest and Enzyme
          </h2>
          <div
            className="row"
          >
            <p>
              Loading...
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              React Testing with Jest and Enzyme
            </h2>,
            <div
              className="row"
            >
              <p>
                Loading...
              </p>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "React Testing with Jest and Enzyme",
            },
            "ref": null,
            "rendered": "React Testing with Jest and Enzyme",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Loading...
              </p>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading...",
              },
              "ref": null,
              "rendered": "Loading...",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
=======
exports[`<App /> renders correctly using shallow 1`] = `ShallowWrapper {}`;
>>>>>>> e69a2d2 (updated test snapshot for render and shallow)
